tfhe/src/integer/server_key/radix/bitwise_op.rs

```diff
@@ -601,4 +601,137 @@ impl ServerKey {
+    /// assert_eq!(dec_result, msg >= 10 && msg <= 15);
+    /// ```
+    pub fn boolean_bitand(&self, lhs: &BooleanBlock, rhs: &BooleanBlock) -> BooleanBlock {
+        let result = self.key.bitand(&lhs.0, &rhs.0);
+        BooleanBlock::new_unchecked(result)
+    }
+
+    pub fn boolean_bitand_assign(&self, lhs: &mut BooleanBlock, rhs: &BooleanBlock) {
+        self.key.bitand_assign(&mut lhs.0, &rhs.0);
+    }
+
+    /// Computes homomorphically a bitor between two boolean ciphertexts
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use tfhe::integer::gen_keys_radix;
+    /// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+    ///
+    /// // Generate the client key and the server key:
+    /// let num_blocks = 4;
+    /// let (cks, sks) = gen_keys_radix(PARAM_MESSAGE_2_CARRY_2_KS_PBS, num_blocks);
+    ///
+    /// let msg = 14u8;
+    ///
+    /// let mut ct = cks.encrypt(msg);
+    ///
+    /// let ct_is_ge = sks.scalar_ge_parallelized(&ct, 10);
+    /// let ct_is_le = sks.scalar_le_parallelized(&ct, 15);
+    ///
+    /// let ct_final_condition = sks.boolean_bitor(&ct_is_ge, &ct_is_le);
+    ///
+    /// // Decrypt:
+    /// let dec_result = cks.decrypt_bool(&ct_final_condition);
+    /// assert_eq!(dec_result, msg >= 10 || msg <= 15);
+    /// ```
+    pub fn boolean_bitor(&self, lhs: &BooleanBlock, rhs: &BooleanBlock) -> BooleanBlock {
+        let result = self.key.bitor(&lhs.0, &rhs.0);
+        BooleanBlock::new_unchecked(result)
+    }
+
+    pub fn boolean_bitor_assign(&self, lhs: &mut BooleanBlock, rhs: &BooleanBlock) {
+        self.key.bitor_assign(&mut lhs.0, &rhs.0);
+    }
+
+    /// Computes homomorphically a bitxor between two boolean ciphertexts
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use tfhe::integer::gen_keys_radix;
+    /// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+    ///
+    /// // Generate the client key and the server key:
+    /// let num_blocks = 4;
+    /// let (cks, sks) = gen_keys_radix(PARAM_MESSAGE_2_CARRY_2_KS_PBS, num_blocks);
+    ///
+    /// let msg = 14u8;
+    ///
+    /// let mut ct = cks.encrypt(msg);
+    ///
+    /// let ct_is_ge = sks.scalar_ge_parallelized(&ct, 10);
+    /// let ct_is_le = sks.scalar_le_parallelized(&ct, 15);
+    ///
+    /// let ct_final_condition = sks.boolean_bitxor(&ct_is_ge, &ct_is_le);
+    ///
+    /// // Decrypt:
+    /// let dec_result = cks.decrypt_bool(&ct_final_condition);
+    /// assert_eq!(dec_result, ((msg >= 10) ^ (msg <= 15)));
+    /// ```
+    pub fn boolean_bitxor(&self, lhs: &BooleanBlock, rhs: &BooleanBlock) -> BooleanBlock {
+        let result = self.key.bitxor(&lhs.0, &rhs.0);
+        BooleanBlock::new_unchecked(result)
+    }
+
+    pub fn boolean_bitxor_assign(&self, lhs: &mut BooleanBlock, rhs: &BooleanBlock) {
+        self.key.bitxor_assign(&mut lhs.0, &rhs.0);
+    }
+
+    /// Computes homomorphically a bitnot between boolean blocks
```

Comment:
between boolean blocks ? 😅 sorry maybe I should have written everything down

Reply:
🥶